cmd/eupf/pfcp_session_handlers.go

```diff
@@ -573,16 +573,19 @@ func applyDownlinkPDR(pdi []*ie.IE, spdrInfo SPDRInfo, pdrId uint16, session *Se
 		if ueIp.IPv4Address != nil {
 			// net.IP is a trap, it needs to be copied, otherwise it will be overwritten by next packet.
 			spdrInfo.Ipv4 = cloneIP(ueIp.IPv4Address)
+			session.PutDownlinkPDR(uint32(pdrId), spdrInfo)
+			if err := mapOperations.PutPdrDownLink(spdrInfo.Ipv4, spdrInfo.PdrInfo); err != nil {
+				log.Printf("Can't put downlink PDR: %s", err.Error())
+			}
 		} else {
 			log.Print("WARN: No IPv4 address")
 		}
 		if ueIp.IPv6Address != nil {
-			log.Print("WARN: UE IPv6 not supported yet, ignoring")
-			return fmt.Errorf("IPv6 not supported")
-		}
-		session.PutDownlinkPDR(uint32(pdrId), spdrInfo)
-		if err := mapOperations.PutPdrDownLink(spdrInfo.Ipv4, spdrInfo.PdrInfo); err != nil {
-			log.Printf("Can't put downlink PDR: %s", err.Error())
+			spdrInfo.Ipv6 = cloneIP(ueIp.IPv6Address)
+			session.PutDownlinkPDR(uint32(pdrId), spdrInfo)
+			if err := mapOperations.PutPdrDownLink(spdrInfo.Ipv6, spdrInfo.PdrInfo); err != nil {
```

Comment:
Don't we need ip6-specific version of `PutPdrDownLink` here?

Reply:
Indeed, I've forgot a postfix.